src/rbr/RBR.java

```diff
@@ -134,22 +134,28 @@ private void adjustRegions(Vertex sw) {
         ArrayList<Region> newRegions = new ArrayList<>();
         ArrayList<Region> regionsToBeRemoved = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            Set<Vertex> outsiders = currentRegion.outsiders();
-            if(outsiders.isEmpty())
-                continue;
-            Range outsidersBox = TopologyKnowledge.box(outsiders);
-            Set<Vertex> trulyDestinationsInOutsidersRange = currentRegion.destinationsIn(outsidersBox);
-
+            newRegions.addAll(adjustRegions(currentRegion));
             regionsToBeRemoved.add(currentRegion);
-            ArrayList<Region> regionsToAdd = splitRegionExcludingOutsiders(currentRegion, outsidersBox);
-            newRegions.addAll(regionsToAdd);
-            // use others routers to make others regions
-            newRegions.addAll(makeRegions(trulyDestinationsInOutsidersRange, currentRegion.inputPorts(), currentRegion.outputPorts()));
         }
         regionsForVertex.get(sw).removeAll(regionsToBeRemoved);
         regionsForVertex.get(sw).addAll(newRegions);
     }
 
+    private ArrayList<Region> adjustRegions(Region r) {
```

Comment:
As you are 'getting' something, a better name to that method could be 'adjustedRegions' or 'adjustedRegionsFrom'.